examples/api/lib/material/navigation_drawer/navigation_drawer.0.dart

```diff
@@ -0,0 +1,171 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for Scaffold.drawer
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  static const String _title = 'Flutter Code Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: _title,
+      home: MyStatefulWidget(),
+    );
+  }
+}
+
+class MyStatefulWidget extends StatefulWidget {
+  const MyStatefulWidget({super.key});
+
+  @override
+  State<MyStatefulWidget> createState() => _MyStatefulWidgetState();
+}
+
+class _MyStatefulWidgetState extends State<MyStatefulWidget> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+  int screenIndex = 0;
+  void handleScreenChanged(int selectedScreen) {
+    setState(() {
+      screenIndex = selectedScreen;
+    });
+  }
+
+  void _openDrawer() {
+    _scaffoldKey.currentState!.openEndDrawer();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final Widget myDrawer = NavigationDrawer(
+      onDestinationSelected: handleScreenChanged,
+      selectedIndex: screenIndex,
+      children: <Widget>[
+        Padding(
+          padding: const EdgeInsets.fromLTRB(28, 16, 16, 10),
+          child: Text(
+            'Header',
+            style: Theme.of(context).textTheme.titleSmall,
+          ),
+        ),
+        ...navigationDestinationDatas
```

Comment:
Since this is where all the action is it might be helpful to call attention to it with a comment. For example:
```
// Create on NavigationDestination for each element of navigationDestinationDatas
```